protocol/app/prepare/prepare_proposal.go

```diff
@@ -52,8 +55,24 @@ func PrepareProposalHandler(
 	txConfig client.TxConfig,
 	bridgeKeeper PrepareBridgeKeeper,
 	clobKeeper PrepareClobKeeper,
-	pricesKeeper PreparePricesKeeper,
+	pricesKeeper prices.PricesKeeper,
```

Comment:
Why are you taking the prices keeper here instead of just using the price update generator interface directly? Not sure what wrapping the initialization func gets us.

Reply:
This j helped reduce the diff, so that the function signature of their original PrepareProposal constructor remained the same